src/main/scala/is/hail/io/bgen/BgenInputFormat.scala

```diff
@@ -1,13 +1,107 @@
 package is.hail.io.bgen
 
-import is.hail.io.IndexedBinaryInputFormat
+import is.hail.utils._
+import is.hail.io._
+import org.apache.hadoop.fs.Path
 import org.apache.hadoop.io.LongWritable
 import org.apache.hadoop.mapred._
 
 class BgenInputFormatV12 extends IndexedBinaryInputFormat[BgenRecordV12] {
   override def getRecordReader(split: InputSplit, job: JobConf, reporter: Reporter): RecordReader[LongWritable,
     BgenRecordV12] = {
     reporter.setStatus(split.toString)
-    new BgenBlockReaderV12(job, split.asInstanceOf[FileSplit])
+    new BgenBlockReaderV12(job, split.asInstanceOf[BgenV12InputSplit])
   }
+
+  override def getSplits(job: JobConf, numSplits: Int): Array[InputSplit] = {
+    val splits = super.getSplits(job, numSplits)
+    splits.flatMap { x =>
+      val split = x.asInstanceOf[FileSplit]
+      val path = split.getPath
+      val positions = job.get(LoadBgen.includedVariantsPositionsHadoopPrefix + path)
+      val indices = job.get(LoadBgen.includedVariantsIndicesHadoopPrefix + path)
+      assert(positions == null && indices == null ||
+        positions != null && indices != null)
+      if (positions != null) {
+        val decodedPositions = LoadBgen.decodeLongs(positions)
+        log.info(s"variant filters found for $path: ${decodedPositions.mkString("[", ",", "]")}")
```

Comment:
Is this going to cause a performance issue if you are logging too many positions?

Reply:
I suppose it's not the smartest thing to do but it's useful debugging information....

In practice, this is used for dramatic filters so this is on the order of 10s of thousands of longs, not too big.

Kinda inclined to keep it as long as this is a private parameter.

Reply:
Actually, at this point it's just depending on decode longs being correct. I'll remove it.